packages/types/src/types.ts

```diff
@@ -74,3 +75,28 @@ export type PreparserExtensionFromHeadmatter = (headmatter: any, exts: SlidevPre
 export type RenderContext = 'slide' | 'overview' | 'presenter' | 'previewNext'
 
 export type LoadedSnippets = Record<string, string>
+
+export type ClicksElement = Element | string
+
+export type ClicksFlow = Map<ClicksElement, number>
+
+export interface ResolvedClicksInfo {
+  max: number
+  flowSize: number
+  isCurrent?: ComputedRef<boolean>
+  isActive?: ComputedRef<boolean>
+  shows?: ComputedRef<boolean>
+}
```

Comment:
I think to me `flow` is still an a bit ambgious word - maybe we could name the type as `absolute` / `relative` everywhere?

The interface could be changed to:

```suggestion
export interface ResolvedClicksInfo {
  type: 'absolute' | 'relative'
  max: number
  delta: number
  resolved: number | [number, number]
  isCurrent?: ComputedRef<boolean>
  isActive?: ComputedRef<boolean>
  isShown?: ComputedRef<boolean>
}
```

Reply:
I think it is a good idea to use "relative"&"absolute" instead of "flow".

Maybe it is better to change the interface to:

```ts
export interface ResolvedClicksInfo {
  max: number
  relativeDelta: number    // For absolute element, it is 0
  isCurrent?: ComputedRef<boolean>
  isActive?: ComputedRef<boolean>
  isShown?: ComputedRef<boolean>
}
```

Reason:

- Maybe there will be clicks elements that can't be resolved into one click or one range.
- The `delta` of absolute elements will not be used by the click system.
- A new `if-else` statement will be added if we use `ResolvedClicksInfo.type` to distinguish two types.

Is the new one OK?

Reply:
Sounds good to me!